Common/app/ota/ota_update_task.c

```diff
@@ -1218,10 +1211,11 @@ void vOTAUpdateTask( void * pvParam )
         }
     }
 
-    if( xResult == pdPASS )
+    if( ( xResult == pdPASS ) &&
+        ( xMQTTAgentHandle != NULL ) )
     {
         xMQTTStatus = MqttAgent_SubscribeSync( xMQTTAgentHandle,
-                                               OTA_JOB_UPDATE_TOPIC_FILTER,
+                                               OTA_JOB_NOTIFY_TOPIC_FILTER,
```

Comment:
You dont need to subscribe for job notify topic manually. OTA Agent will invoke a callback for subscription to job notify filter.